firebase-provider/src/main/java/com/busybusy/firebase_provider/FirebaseProvider.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2018 - 2022 busybusy, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.busybusy.firebase_provider
+
+import com.google.firebase.analytics.FirebaseAnalytics
+import com.busybusy.analyticskit_android.AnalyticsKitProvider.PriorityFilter
+import com.busybusy.analyticskit_android.AnalyticsKitProvider
+import com.busybusy.analyticskit_android.AnalyticsEvent
+import android.os.Bundle
+import java.io.Serializable
+import java.lang.IllegalStateException
+import java.text.DecimalFormat
+
+/**
+ * Provider that facilitates reporting events to Firebase Analytics.
+ * We recommend using the [FirebaseAnalytics.Param] names for attributes as much as possible due to the restriction on the number of
+ * custom parameters in Firebase.
+ *
+ * @see [Log Events](https://firebase.google.com/docs/analytics/android/events)
+ * @see [Custom-parameter reporting](https://support.google.com/firebase/answer/7397304?hl=en&ref_topic=6317489)
+ *
+ * @property firebaseAnalytics the initialized [FirebaseAnalytics] instance associated with the application
+ * @property priorityFilter the [PriorityFilter] to use when evaluating if an event should be sent to this provider's platform.
+ *                          By default, this provider will log all events regardless of priority.
+ */
+class FirebaseProvider(
+    private val firebaseAnalytics: FirebaseAnalytics,
+    private var priorityFilter: PriorityFilter = PriorityFilter { true },
+) : AnalyticsKitProvider {
+
+    private val timedEvents: MutableMap<String, AnalyticsEvent> by lazy { mutableMapOf() }
```

Comment:
Still using lazy initialization of these map properties but this Kotlin version is much more satisfying! :grinning: